packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -947,7 +955,7 @@ Future<T> showCupertinoDialog<T>({
   return showGeneralDialog(
     context: context,
     barrierDismissible: false,
-    barrierColor: _kModalBarrierColor,
+    barrierColor: CupertinoDynamicColor.resolve(_kModalBarrierColor, context),
```

Comment:
The caveat of doing this is, the color won't change unless the dialog is dismissed then shown again.

Reply:
Seems OK.  Even if we recreated showGeneralDialog's _DialogRoute, it still wouldn't be easy to get the scrim to redraw when the platform's brightness changed.